src/librustc_errors/registry.rs

```diff
@@ -1,16 +1,39 @@
+use crate::registry::RegistryResult::HasLongDescription;
 use rustc_data_structures::fx::FxHashMap;
 
+pub enum RegistryResult {
+    InvalidErrorCode,
+    NoLongDescription,
+    HasLongDescription(&'static str),
+}
+
+impl RegistryResult {
+    pub fn has_value(&self) -> bool {
+        match self {
+            HasLongDescription(_s) => true,
+            _ => false,
+        }
+    }
+}
+
 #[derive(Clone)]
 pub struct Registry {
-    descriptions: FxHashMap<&'static str, &'static str>,
+    long_descriptions: FxHashMap<&'static str, Option<&'static str>>,
 }
 
 impl Registry {
-    pub fn new(descriptions: &[(&'static str, &'static str)]) -> Registry {
-        Registry { descriptions: descriptions.iter().cloned().collect() }
+    pub fn new(long_descriptions: &[(&'static str, Option<&'static str>)]) -> Registry {
+        Registry { long_descriptions: long_descriptions.iter().cloned().collect() }
     }
 
-    pub fn find_description(&self, code: &str) -> Option<&'static str> {
-        self.descriptions.get(code).cloned()
+    pub fn find_description(&self, code: &str) -> RegistryResult {
+        if !self.long_descriptions.contains_key(code) {
+            return RegistryResult::InvalidErrorCode;
+        }
+        let entry = self.long_descriptions.get(code).unwrap(); // use unwrap here because we checked the key exists already
```

Comment:
Nit: let's move the comment to it's own line above the statement; getting a bit too long to the right.